streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -208,6 +214,26 @@ public synchronized void close() {
     }
 
     /**
+     * Cleans up local state store directory ({@code state.dir}), by deleting all data with regard to the application-id.
+     * <p>
+     * May only be called before instance is started or after instance is closed.
+     *
+     * @throws IllegalStateException if instance is currently running
+     */
+    public void localCleanUp() {
+        if (state == RUNNING) {
+            throw new IllegalStateException("Cannot clean up while running.");
+        }
+
+        final String localApplicationDir = this.config.getString(StreamsConfig.STATE_DIR_CONFIG)
+                + File.separator
+                + this.config.getString(StreamsConfig.APPLICATION_ID_CONFIG);
+
+        log.debug("Clean up local Kafka Streams data in {}", localApplicationDir);
```

Comment:
Perhaps:
```java
log.debug("Removing local Kafka Streams application data in {} for application {}", localApplicationDir, this.config.getString(StreamsConfig.APPLICATION_ID_CONFIG));
```

Because we're actually removing files, we're not "cleaning" (which some might interpret as modifying but not deleting files).  Also, though the app ID is somewhat included in the app dir, I think we should be super clear what exactly is being removed. ("Oh my god, did this just remove my data in Kafka??)  (I know the "local" was already there...)